notebooks/ima_viewer/ima_viewer.ipynb

```diff
@@ -0,0 +1,587 @@
+       "            </div>\n",
+       "        "
+      ],
+      "text/plain": [
+       "Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot = make_interactive_plot(filepath, ext_type='SCI')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd09c20e",
+   "metadata": {},
+   "source": [
+    "<a id=\"save\"></a>\n",
+    "### 2.3 Save files\n",
+    "\n",
+    "Running the cell below will generate options for saving your IMA as frames, a GIF, or both. The limits, colormap, scaling, etc. will reflect whatever is currently set above. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "b7c8fc61",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "d74839fb8783444eab64c637fae34b84",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "VBox(children=(HBox(children=(VBox(children=(HBox(children=(HTML(value='<style>p{word-wrap: break-word}</style…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "gif_setup = gif_settings_widgets(plot)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7d517e4",
+   "metadata": {},
+   "source": [
+    "When the IMA viewer and the widget settings are to your liking, run the following two cells to \n",
```

Comment:
I tried to break the widgets as much as I could. You can choose to fix all of these or just define where these break:

1. "Make GIF from" needs to be chosen.
2. You cannot save individual frames without choosing an option in "Making GIF from"
3. Making a GIF doesn't default to PNG when nothing is chosen. "Do you want to save ..." should be changed to "Which individual frames would you like to save?"
4. It always saves the file extension used for making GIFs, regardless of which individual frames you want to save.